### ISSUE TITLE
Add API to inject histories

### ISSUE THREAD

**Issue by chrisvxd** [measuredco/puck#492]
## Problem

Some users may wish to record history in their database, and inject the history back into Puck.

**Proposal 2 is preferable**

### Option 1 - Add `initialHistories` array prop

Add an `initialHistories` array prop to the Puck component

```tsx
<Puck initialHistories={[]} />
```

### Option 2 - Add `initialHistory` object prop

Add an `initialHistory` object prop to the Puck component

```tsx
<Puck initialHistory={{ histories: [], index: 0 }} />
```

## Considerations

* The `usePuck` histories are not currently documented
* #160 will impact implementation

**Comment by mkilpatrick** [comment]
You may want to take in the currentHistoryIndex as well so users can get back to exactly where they left off.

**Comment by chrisvxd** [comment]
Great suggestion @mkilpatrick - added an option 2.

**Comment by mkilpatrick** [comment]
Also noting that this may be affected by the history diff changes if https://github.com/measuredco/puck/issues/160 is completed.

**Comment by chrisvxd** [comment]
Added as a consideration, and marked this as ready for someone to pick up!

**Comment by chrisvxd** [comment]
Closed by #505
